src/app/lib/main.dart

```diff
@@ -1,20 +1,51 @@
 import 'package:app/access/dad_jokes/dad_jokes_repository.dart';
+import 'package:app/access/environment/environment_repository.dart';
 import 'package:app/access/dad_jokes/favorite_dad_jokes_repository.dart';
 import 'package:app/access/diagnostics/diagnostics_repository.dart';
 import 'package:app/app.dart';
 import 'package:app/business/dad_jokes/dad_jokes_service.dart';
 import 'package:app/business/diagnostics/diagnostics_service.dart';
+import 'package:app/business/environment/environment_manager.dart';
+import 'package:app/presentation/configuration/environment_configuration.dart';
 import 'package:dio/dio.dart';
 import 'package:flutter/material.dart';
+import 'package:flutter_dotenv/flutter_dotenv.dart';
 import 'package:get_it/get_it.dart';
 
-void main() {
+Future<void> main() async {
+  WidgetsFlutterBinding.ensureInitialized();
+  await _registerAndLoadEnvironment();
+
   _registerHttpClient();
   _registerRepositories();
   _registerServices();
+
   runApp(const App());
 }
 
+Future _registerAndLoadEnvironment() async {
+  // Register environment services in the IoC.
+  GetIt.I.registerSingleton(EnvironmentRepository());
+  GetIt.I.registerSingleton(dotenv);
```

Comment:
Wondering if we should still inject this. I think this variable is static and do we gain anything by putting it in this container. If I compare it too shared preference, we just use it straight in the service may be it's better to do it your way, just unsure.

Reply:
Yeah I had the refllex of .NET where everything is in IoC, but maybe to be consistent here I could just use the static reference directly